doc/macro_reference.qbk

```diff
@@ -1081,6 +1082,33 @@ that are not yet supported by a particular compiler or library.
 
 [endsect]
 
+[section Macros that allow C++20 features to be used in previous standard versions]
+
+[table
+[[macro][Description]]
+[[`BOOST_IS_CONSTANT_EVALUATED()`][Evaluated to `true` when the current context is `constexpr` and false otherwise.
```

Comment:
beginner question: why is this a macro?

I think boost::is_constant_evaluated() would be a bit nicer, I presume this is for consistency with other stuff in this repo?

Reply:
We would need to include <type_traits> to cover that, and I'd very much like to NOT do that.

Explanation: if it's a compiler about which we know nothing about, ie we don't know what it calls the compiler intrinsic that implements this, then we would need to be able to forward to std::is_constant_evaluated if it's presence is advertised in <version>.

As it stands, it's up to users of these macros to include <type_traits> and everyone else can do without if they wish and only pay for what they use.

We could of course put a real function into Boost.Core.

Reply:
Ah, I see, I was naively assuming that that feature macro is predefined, now I see it is in `<type_traits>`. Shame.